pysite/views/api/bot/user.py

```diff
@@ -0,0 +1,45 @@
+# coding=utf-8
+
+from flask import jsonify, request
+
+from pysite.base_route import APIView, DBViewMixin
+from pysite.constants import ErrorCodes
+from pysite.decorators import valid_api_key
+
+REQUIRED_KEYS = [
+    "user_id",
+    "role"
+]
+
+
+class TagView(APIView, DBViewMixin):
+    path = "/user"
+    name = "user"
+    table_name = "users"
+    table_primary_key = "user_id"
+
+    @valid_api_key
+    def post(self):
+        data = request.get_json()
+
+        if not isinstance(data, list):
+            data = [data]
+
+        for user in data:
+            if not all(k in user for k in REQUIRED_KEYS):
```

Comment:
how about we repurpose the `valid_api_key` decorator to be `valid_api_request` instead, then we make `required_keys` (maybe rename to `required_params`?) be a class attribute, and then the decorator can check if they're present.

That way we only have to define the required params and all the validation is handled automatically.

Reply:
My plan is to have API key management via the database, so it doesn't make sense to supply sets of API keys here.

Reply:
I think that might be a little complex - what do we do if a list is expected?